### PR TITLE
Add email body template & Optimize notification setting UI

### PR DESCRIPTION
**Pull request by zhongnansu**
*Issue #, if available:*

*Description of changes:*
- update notification settings UI and flow behind in create report definition 
image

- Add email template 
image

- update notification setting UI in report details page
- Add better logging and error handling for email

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/public/components/report_definitions/edit/edit_report_definition.tsx [hunk 1]
```diff
@@ -97,7 +97,6 @@ export function EditReportDefinition(props) {
 
   const callUpdateAPI = async (metadata) => {
     const { httpClient } = props;
-
```
Comment (davidcui-amzn) [inline kibana-reports/public/components/report_definitions/edit/edit_report_definition.tsx:100]: Can we remove this file from the PR?

File: kibana-reports/server/model/index.ts [hunk 2]
```diff
@@ -143,13 +142,12 @@ export const channelSchema = schema.object({
     }),
     { minSize: 1 }
   ),
+  // TODO: consider add this field next to url-related fields.
+  // Need this to build the links in email
+  origin: schema.uri(), //e.g. https://xxxxx.com
```
Comment (davidcui-amzn) [inline kibana-reports/server/model/index.ts:147]: Do we need input validation for this?

File: kibana-reports/server/routes/lib/deliverReport.ts [hunk 3]
```diff
@@ -49,37 +47,73 @@ export const deliverReport = async (
...
+      notificationClient,
+      'notification.send',
+      {
+        body: deliveryBody,
+      },
+      isScheduledTask
+    );
+    /**
+     * notification plugin response example:
+     * {
```
Comment (davidcui-amzn) [inline kibana-reports/server/routes/lib/deliverReport.ts:88]: Can we remove this comment block?

**Review by davidcui-amzn — APPROVED** [review]
LGTM - please address comments before merging

File: kibana-reports/server/model/index.ts (hunk 2 above)
Comment (zhongnansu) [reply kibana-reports/server/model/index.ts:147]: the `.uri()` will check whether it is valid uri in this format `https://xxxxx.com`

File: kibana-reports/public/components/report_definitions/edit/edit_report_definition.tsx (hunk 1 above)
Comment (zhongnansu) [reply kibana-reports/public/components/report_definitions/edit/edit_report_definition.tsx:100]: do you mean `edit_report_definition.tsx`

Comment (davidcui-amzn) [reply kibana-reports/public/components/report_definitions/edit/edit_report_definition.tsx:100]: Yeah- it looks like the only change is an empty line removal?

Comment (zhongnansu) [reply kibana-reports/public/components/report_definitions/edit/edit_report_definition.tsx]: done

**Review by anirudha — APPROVED** [review]

**Pull request merged**